src/analysis/seminorm.lean

```diff
@@ -708,6 +714,60 @@ section normed_field
 variables [normed_field 𝕜] [add_comm_group E] [module 𝕜 E] (p : seminorm 𝕜 E) {A B : set E}
   {a : 𝕜} {r : ℝ} {x : E}
 
+lemma bdd_below_range_add (x : E) (p q : seminorm 𝕜 E) :
+  bdd_below (range (λ (u : E), p u + q (x - u))) :=
+by { use 0, rintro _ ⟨x, rfl⟩, exact add_nonneg (p.nonneg _) (q.nonneg _) }
+
+/- can't put it with `has_sup` because it requires much stronger properties on 𝕜, E. -/
+noncomputable instance : has_inf (seminorm 𝕜 E) :=
+{ inf := λ p q,
+  { to_fun := λ x, ⨅ u : E, p u + q (x-u),
+    triangle' := λ x y, begin
+      rw ← sub_le_iff_le_add,
+      apply le_cinfi, intro u,
+      rw sub_le,
+      apply le_cinfi, intro v,
+      rw sub_le_iff_le_add,
+      apply cinfi_le_of_le (bdd_below_range_add _ _ _) (v+u), dsimp only,
+      convert add_le_add (p.triangle v u) (q.triangle (y-v) (x-u)) using 1,
+      { have h : x + y - (v + u) = y - v + (x - u), { abel }, { rw h } },
+      { abel },
+    end,
+    smul' := λ a x, begin
+      obtain ha | ha := (norm_nonneg a).eq_or_lt,
+      { rw ← ha, ring_nf, apply le_antisymm,
+        { apply cinfi_le_of_le (bdd_below_range_add _ _ _) (0:E),
+          simp only [seminorm.zero, sub_zero, zero_add], rw q.smul, rw ← ha, simp only [zero_mul]},
+        { apply le_cinfi, intro, exact add_nonneg (p.nonneg _) (q.nonneg _) } },
+      { apply le_antisymm,
+        { rw ← div_le_iff' ha,
+          apply le_cinfi, intro u,
+          rw div_le_iff' ha,
+          apply cinfi_le_of_le (bdd_below_range_add _ _ _) (a • u),
+          dsimp, rw ← smul_sub, rw p.smul, rw q.smul, rw mul_add },
```

Comment:
```suggestion
          dsimp, rw [← smul_sub, p.smul, q.smul, mul_add] },
```
same for other rewrites

Reply:
the `dsimp` should also not be there, but this is an artifact of the `bdd_below_range_add` lemma and will probably disappear automatically as soon as `bdd_below_range_add` is gone